source/ast/symbols/MemberSymbols.cpp

```diff
@@ -2119,16 +2124,18 @@ std::span<const Expression* const> NetAliasSymbol::getNetReferences() const {
     auto syntax = getSyntax();
     SLANG_ASSERT(scope && syntax);
 
-    // TODO: there should be a global check somewhere that any given bit
-    // of a net isn't aliased to the same target signal bit multiple times.
     bitwidth_t bitWidth = 0;
     bool issuedError = false;
     SmallVector<const Expression*> buffer;
     ASTContext context(*scope, LookupLocation::after(*this),
                        ASTFlags::NonProcedural | ASTFlags::NotADriver);
     NetAliasVisitor visitor(context);
+    // In order to save the leftmost alias
+    SmallMap<const ValueSymbol*, const Expression*, 2> leftAlias;
+
+    const auto nets = syntax->as<NetAliasSyntax>().nets;
 
-    for (auto exprSyntax : syntax->as<NetAliasSyntax>().nets) {
+    for (auto exprSyntax : nets) {
         auto& netRef = Expression::bind(*exprSyntax, context);
         if (!netRef.requireLValue(context))
```

Comment:
And then here pass your new AssignFlag and then you won't need the additional addDriver call below I think.

Reply:
done